cmd/ci-images-mirror/main.go

```diff
@@ -84,12 +96,41 @@ func main() {
 	}
 
 	ctx := controllerruntime.SetupSignalHandler()
-
-	inClusterConfig, err := rest.InClusterConfig()
+	inClusterConfig, err := util.LoadClusterConfig()
 	if err != nil {
 		logrus.WithError(err).Fatal("Failed to load in-cluster config")
 	}
 
+	eventCh := make(chan fsnotify.Event)
+	errCh := make(chan error)
+	universalSymlinkWatcher := &agents.UniversalSymlinkWatcher{
+		EventCh:   eventCh,
+		ErrCh:     errCh,
+		WatchPath: opts.releaseRepoGitSyncPath,
+	}
+	configAgentOption := func(opt *agents.ConfigAgentOptions) {
+		opt.UniversalSymlinkWatcher = universalSymlinkWatcher
+	}
+	ciOperatorConfigPath := filepath.Join(opts.releaseRepoGitSyncPath, config.CiopConfigInRepoPath)
+
+	configErrCh := make(chan error)
+	ciOPConfigAgent, err := agents.NewConfigAgent(ciOperatorConfigPath, configErrCh, configAgentOption)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to construct ci-operator config agent")
+	}
+	go func() { logrus.Fatal(<-configErrCh) }()
+
+	registryErrCh := make(chan error)
+	registryAgentOption := func(opt *agents.RegistryAgentOptions) {
+		opt.UniversalSymlinkWatcher = universalSymlinkWatcher
+	}
+	stepConfigPath := filepath.Join(opts.releaseRepoGitSyncPath, config.RegistryPath)
+	registryConfigAgent, err := agents.NewRegistryAgent(stepConfigPath, registryErrCh, registryAgentOption)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to construct registryAgent")
+	}
+	go func() { logrus.Fatal(<-registryErrCh) }()
```

Comment:
I think it would be better to check both of the channels in the same go routine since the plan here is to exit on error. Otherwise, we need to make sure that the other go routine stops before exiting.